R/dictionary.R

```diff
@@ -269,14 +269,25 @@ vec_in <- function(needles, haystack) {
 #'   as_tibble(vec_split(mtcars, mtcars[c("vs", "am")]))
 #' }
 vec_split <- function(x, by) {
-  # TODO: optimise to avoid creating the dictionary twice and to avoid split()
-  keys <- vec_unique(by)
+
   if (vec_size(x) != vec_size(by)) {
     abort("`x` and `by` must have same size")
   }
 
-  idx <- unname(split(vec_seq_along(x), vec_duplicate_id(by)))
-  vals <- new_list_of(map(idx, vec_slice, x = x), vec_type(x))
+  ki <- vec_duplicate_split(by)
+
```

Comment:
Can you group these three lines together please? Each line as a paragraph is a bit too much whitespace.

Reply:
✅